users/rLapz/old/client.c

```diff
@@ -0,0 +1,212 @@
+/* @author rLapz <arthurlapz@gmail.com>
+ * @license MIT
+ *
+ * This is part of simple_file_transfer
+ * https://github.com/teainside/simple_file_transfer
+ *
+ * NOTE: true = 0, false = 1
+ */
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <arpa/inet.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+
+/* local include */
+#include "data_structure.h"
+
+/* macros */
+#define TRUE	0
+#define FALSE	1
+
+/* function declarations */
+static int inet_handler(const char* address,
+		uint16_t port, const char* filename);
+static char * get_basename(const char *path);
+
+/* global variable declarations */
+/* anyway, there is no global variable here, yay! */
+
+
+int
+main(int argc, char *argv[])
+{
+	if (argc < 4) {
+		printf("Usage:\n\t%s [address] [port] [file_target]\n", argv[0]);
+		return EXIT_FAILURE;
+	}
+
+	return inet_handler(argv[1], (uint16_t)atoi(argv[2]), argv[3]);
+}
+
+/* function for handling internet connection
+ * and file i/o operations.
+ * sadly, not written into separated function, yet!
+ */
+static int
+inet_handler(const char* address,
+		uint16_t port, const char* filename)
+{
+	struct stat s_file	= {0};
+	struct sockaddr_in server = {0};
+
+	short int is_error	= FALSE;
+	int client_fd		= 0;
+	int file_fd		= 0;
+	ssize_t read_bytes	= 0;
+	ssize_t send_bytes	= 0;
+	size_t sent_bytes	= 0;
+	/* why heap? because I need flexible memory size, though. */
+	char *data_arena	= NULL;
+	char *data_arena_tmp	= NULL;
+	packet *packet_data	= NULL;
+	char *basename		= NULL;
+	
+
+	data_arena = calloc(sizeof(packet), sizeof(packet));
+	if (data_arena == NULL) {
+		perror("Allocation data_arena");
+		is_error = TRUE;
+		return EXIT_FAILURE;
+	}
+	packet_data = (packet*)data_arena;
+
+	/* file checking */
+	if (stat(filename, &s_file) < 0) {
+		perror(filename);
+		is_error = TRUE;
+		goto cleanup;
+	}
+
+	/*
+	 * set file properties
+	 */
+
+	/* get basename of a file */
+	basename = get_basename(filename);
+	if (strlen(basename) == 0) {
+		fprintf(stderr, "File/path invalid\n");
+		is_error = TRUE;
+		goto cleanup;
+	}
+
+	strncpy(packet_data->filename, basename, sizeof(packet_data->filename));
+	packet_data->filename_len = (uint8_t)strlen(basename);
+	packet_data->file_size = (uint64_t)s_file.st_size;
+
+	/* print file properties */
+	puts("\nFile info: ");
+	printf(" -> File name\t\t: %s\n", packet_data->filename);
+	printf(" -> File name length:\t: %d\n", packet_data->filename_len);
+	printf(" -> File size\t\t: %zu bytes\n", packet_data->file_size);
+	printf(" -> Send to\t\t: %s:%d\n", address, port);
+
+	/* Create socket */
+	client_fd = socket(AF_INET, SOCK_STREAM, 0);
+	if (client_fd < 0) {
+		perror("Create socket");
+		is_error = TRUE;
+		goto cleanup;
+	}
+
+	/* TCP configuration */
+	server.sin_addr.s_addr = inet_addr(address);
+	server.sin_family = AF_INET;
+	server.sin_port = (uint16_t)htons(port);
```

Comment:
```suggestion
	server.sin_port = htons(port);
```
You don't need a cast, `htons` already returns `uint16_t`.